test/e2e/sources/azureblobstorage/main.go

```diff
@@ -86,227 +82,252 @@ const (
+
+		gvr := sourceAPIVersion.WithResource(sourceResource)
 		srcClient = f.DynamicClient.Resource(gvr).Namespace(ns)
 	})
 
-	Context("an Azure Blob is created and deleted ", func() {
-		var err error // stubbed
+	Context("a source subscribes to events from a blob container", func() {
+		var rgName string
+		var storageAccountName string
+		var blobContainerName string
+
+		subscriptionID := os.Getenv("AZURE_SUBSCRIPTION_ID")
+
+		region := os.Getenv("AZURE_REGION")
+		if region == "" {
+			region = "westus2"
+		}
 
-		When("a blob is created and deleted", func() {
-			BeforeEach(func() {
-				rg = azure.CreateResourceGroup(ctx, subscriptionID, ns, region)
-				_ = azure.CreateEventHubNamespaceOnly(ctx, subscriptionID, ns, region, *rg.Name)
+		ctx := context.Background()
+
+		BeforeEach(func() {
+
+			By("creating a resource group", func() {
+				rg := azure.CreateResourceGroup(ctx, subscriptionID, ns, region)
+				rgName = *rg.Name
 			})
 
-			var src *unstructured.Unstructured
+			By("creating a storage account and Blob container", func() {
+				storCli := azure.CreateStorageAccountsClient(subscriptionID)
+				storageAccount := azure.CreateBlobStorageAccount(ctx, storCli, rgName, region)
+				storageAccountID = *storageAccount.ID
+				storageAccountName = *storageAccount.Name
 
-			BeforeEach(func() {
-				sink = bridges.CreateEventDisplaySink(f.KubeClient, ns)
+				container := azure.CreateBlobContainer(ctx, rgName, *storageAccount.Name, subscriptionID, "e2e")
+				blobContainerName = *container.Name
+			})
 
-				saClient := azure.CreateStorageAccountsClient(subscriptionID)
+			By("creating an Event Hubs namespace", func() {
+				nsName := f.UniqueName
+				_ = azure.CreateEventHubNamespaceOnly(ctx, subscriptionID, nsName, region, rgName)
 
-				// storageaccount name must be alphanumeric characters only and 3-24 characters long
-				saName = strings.Replace(ns, "-", "", -1)
-				saName = strings.Replace(saName, "e2eazureblobstoragesource", "tme2etest", -1)
-				sa = azure.CreateBlobStorageAccount(ctx, saClient, saName, *rg.Name, region)
+				eventHubsNamespaceID = eventHubsNamespaceResourceID(subscriptionID, rgName, nsName)
+			})
 
-				container = azure.CreateBlobContainer(ctx, *rg.Name, sa, subscriptionID, ns)
+			By("creating an event sink", func() {
+				sink = bridges.CreateEventDisplaySink(f.KubeClient, ns)
+			})
 
-				src, err = createSource(srcClient, ns, "test-", sink,
+			By("creating an AzureBlobStorageSource object", func() {
+				src, err := createSource(srcClient, ns, "test-", sink,
 					withServicePrincipal(),
-					withEventTypes([]string{"Microsoft.Storage.BlobCreated", "Microsoft.Storage.BlobDeleted"}),
-					withEventHubNamespace(createEventhubID(subscriptionID, ns)),
-					withStorageAccountID(createStorageAccountID(subscriptionID, ns, saName)),
+					withStorageAccount(storageAccountID),
+					withEventTypes([]interface{}{"Microsoft.Storage.BlobCreated", "Microsoft.Storage.BlobDeleted"}),
```

Comment:
I did that to enable passing non-string values and assert that the spec was rejected, based on the OpenAPI schema.
Now that I think about it, maybe it's not necessary to go that far, otherwise we could as well do that for every field, and that's a rabbit hole.

Reply:
Reverted in [96aec46f..31d4d7be](https://github.com/triggermesh/triggermesh/compare/96aec46fadf34a96282a291d931f35b9a2a411a5..31d4d7be1c4eacca5b9e3465b43e07388aa0541d)